test/functional/test_framework/test_node.py

```diff
@@ -85,24 +85,32 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, timeout_factor, bitc
         self.coverage_dir = coverage_dir
         self.cwd = cwd
         self.descriptors = descriptors
+        self.init_rpc_url = rpc_url(self.datadir, self.index, self.chain, self.rpchost)
         if extra_conf is not None:
             append_config(datadir, extra_conf)
         # Most callers will just need to add extra args to the standard list below.
         # For those callers that need more flexibility, they can just set the args property directly.
         # Note that common args are set in the config file (see initialize_datadir)
         self.extra_args = extra_args
         self.version = version
+
+        # Calculate RPC address to be passed into the command line
+        rpc_addr = self.init_rpc_url.split("http://")[-1].split('@')[-1]
+
         # Configuration for logging is set as command-line args rather than in the bitcoin.conf file.
         # This means that starting a bitcoind using the temp dir to debug a failed test won't
         # spam debug.log.
         self.args = [
             self.binary,
-            "-datadir=" + self.datadir,
-            "-logtimemicros",
-            "-debug",
-            "-debugexclude=libevent",
-            "-debugexclude=leveldb",
-            "-uacomment=testnode%d" % i,
+            "--net=mainnet",
```

Comment:
While this command line parameter should be OK, the default should be mainnet anyway. This has to be replaced with regtest sometime in the future.

Reply:
I added a todo comment saying this should be changed to regtest but left the option in for completeness. Also, made `mainnet` default so the command line option is not strictly necessary.